lib/backend/etcd/etcd.go

```diff
@@ -132,7 +133,7 @@ func (c *EtcdClient) Apply(d *model.KVPair) (*model.KVPair, error) {
 
 // Delete an entry in the datastore.  This errors if the entry does not exists.
 func (c *EtcdClient) Delete(d *model.KVPair) error {
-	key, err := model.KeyToDefaultDeletePath(d.Key)
+	key, parents, err := model.KeyToDefaultDeletePaths(d.Key)
```

Comment:
You know I thought that as I was writing it and then I thought no because... blah... and now I can't think what blah is so I was probably thinking some nonsense or other.  I'll change - I think i'll make the interface much much cleaner!